### PR TITLE
James 1954 Improve Command injection mechanism

### PR DESCRIPTION
**Pull request by chibenwa**
https://github.com/linagora/james-project/issues/668

We should only do  command injection when executing a command in order not to reject valid emails.

I added integration tests for it.

Bonus : we avoid copying 1 time emails data ^^

### CODE REVIEW

File: protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java [hunk 1]
```diff
@@ -251,6 +253,10 @@ protected Request parseRequest(Session session, ByteBuffer buffer) throws Except
         }
         // This should be changed once we move to java6
         String cmdString = new String(line, session.getCharset().name()).trim();
+        if (hasCommandInjection(cmdString.toLowerCase())) {
+            throw new CommandInjectionDetectedException();
+        }
+        System.out.println("CMMMMD string : " + cmdString);
```
Comment (quynhn) [inline protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java]: ?

Comment (chibenwa) [inline protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java]: Arf

**Comment by quynhn** [comment]
`Failed tests: 
  NettyStartTlsSMTPServerTest.startTlsShouldFailWhenFollowedByInjectedCommand`

File: protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java [hunk 2]
```diff
@@ -46,6 +46,9 @@
  *
  */
 public class CommandDispatcher<Session extends ProtocolSession> implements ExtensibleHandler, LineHandler<Session> {
+
+    private static final String STARTTLS = "starttls";
```
Comment (strokyl) [inline protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java]: Could you explain me what kind of exception you are looking for ?

File: protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java [hunk 3]
```diff
@@ -251,6 +254,9 @@ protected Request parseRequest(Session session, ByteBuffer buffer) throws Except
         }
         // This should be changed once we move to java6
         String cmdString = new String(line, session.getCharset().name()).trim();
+        if (hasCommandInjection(cmdString.toLowerCase())) {
```
Comment (mbaechler) [inline protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java]: you are mixing concerns, what about delegating that check ?

Comment (chibenwa) [inline protocols/api/src/main/java/org/apache/james/protocols/api/handler/CommandDispatcher.java]: OutDated code

**Pull request closed without merging**